### PR TITLE
Feature/12922 12911 12913 changes reissuance screen and update network call and business logic

### PR DESCRIPTION
**Pull request by 30mar**
## Description

- Supporting multiple certificates
- Changes to reissuance consent screen
- Update the API for certificate reissuance
- Update business logic to support multiple certificates

**How to test the behaviour:**
1-scan these 3 certificates "mock cloud env" of which 2 are eligible for re issuance:
dcc-series-18-lulu-tanini
2-upon successful reissuance: you will find the 2 reissued certificates removed in the bin and replaced by 2 new ones in the certificates screen

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12922
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12911
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12913

### CODE REVIEW

**Review by KathrinLuecke — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Reissuance/ReissuanceConsent/HealthCertificateReissuanceConsentViewModel.swift [hunk 1]
```diff
@@ -96,15 +112,16 @@ final class HealthCertificateReissuanceConsentViewModel {
...
+			)
+		}
 	}
 
 	func markCertificateReissuanceAsSeen() {
 		certifiedPerson.isNewCertificateReissuance = false
 	}
 
 	func submit(completion: @escaping (Result<Void, HealthCertificateReissuanceError>) -> Void) {
+		/*
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Reissuance/ReissuanceConsent/HealthCertificateReissuanceConsentViewModel.swift:124]: What about this commented code?

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift [hunk 2]
```diff
@@ -1895,6 +1893,7 @@ class HealthCertificateServiceTests: CWATestCase {
 		XCTAssertEqual(service.unseenNewsCount.value, 0)
 	}
 	
+	/*
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift:1896]: What about these tests?

File: src/xcode/ENA/ENA.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -10508,6 +10508,7 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
+				34E03F822823BEF8001A352B /* ccl-configuration.bin in Resources */,
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA.xcodeproj/project.pbxproj:10511]: was the ccl-configuration.bin updated?

**Review by pascal-brause — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA.xcodeproj/project.pbxproj (hunk 3 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA.xcodeproj/project.pbxproj:10511]: Yes

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift (hunk 2 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift]: @pascal-brause  the unit tests have a separate ticket, I'm working on it at the moment and i UN-commented the the code "but the unit tests would not compile now until i finish the other ticket"

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMNewHttpCommunication/DMNHCViewModel.swift [hunk 4]
```diff
@@ -66,8 +66,15 @@ final class DMNHCViewModel {
...
 						return
 					}
-
+					guard let firstPerson = self.healthCertificateService.healthCertifiedPersons.first(where: {
+						$0.healthCertificates.contains(firstVaccinationCertificate)
+					}) else {
+						Log.error("Could not get matching person")
+						return
+					}
+					// TODO update as the new model
```
[the remark this answers is not in the document]
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Developer Menu/Features/DMNewHttpCommunication/DMNHCViewModel.swift]: @pascal-brause Updated

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift [hunk 5]
```diff
@@ -1943,6 +1941,14 @@ class HealthCertificateServiceTests: CWATestCase {
 			completedNotificationRegistration: { }
 		)
 		
+		try service.replaceHealthCertificate(
+			requestCertificates: <#T##[String]#>,
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift:1945]: crazy syntax 🤪

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift (hunk 2 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/__tests__/HealthCertificateServiceTests.swift]: okay, then it is fine for me

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMNewHttpCommunication/DMNHCViewModel.swift [hunk 6]
```diff
@@ -192,6 +199,20 @@ final class DMNHCViewModel {
 	private let appConfiguration: AppConfigurationProviding
 	private let healthCertificateService: HealthCertificateService
 
+	private func submitReissuance(resource: DCCReissuanceResource) {
+		self.restService.load(resource) { result in
+			DispatchQueue.main.async {
+				switch result {
+				case let .success(model):
+					Log.info("DCC Reissuance successfull called.")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMNewHttpCommunication/DMNHCViewModel.swift:207]:
```suggestion
					Log.info("DCC Reissuance successfully called.")
```

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift [hunk 7]
```diff
@@ -265,44 +265,59 @@ class HealthCertificateService: HealthCertificateServiceServable {
...
+								dispatchGroup.leave()
+							}
+						)
+						moveHealthCertificateToBin(certificateToBeRemoved)
+						
+					} else {
+						Log.error("The certified person does not contain the indexed certificate", log: .vaccination)
+					}
+				} else {
+					Log.error("index of certificate to be deleted is out of Bounds", log: .vaccination)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift:314]:
```suggestion
					Log.error("Index of certificate to be deleted is out of bounds", log: .vaccination)
```

**Pull request merged**